tabular/src/autogluon/tabular/models/tabular_nn/tabular_nn_dataset.py

```diff
@@ -319,3 +320,217 @@ def load(cls, file_prefix=""):
         dataset.generate_dataset_and_dataloader(data_list=data_list)
         logger.debug("TabularNN Dataset loaded from files: \n %s \n %s" % (dataobj_file, datalist_file))
         return dataset
+
+
+class TabularPyTorchDataset(torch.utils.data.Dataset):
```

Comment:
I would move all of the pytorch TabularNN classes/functions to separate files from the MXNet ones. This way you can also `import torch` without worrying about this being called when user is only using MXNet NN.

Reply:
And then to minimize copy/paste, I'd move functions used by both MXNet & Pytorch network to a separate helper function file.

Reply:
moving to new file `tabuler_nn_torch.py`